test-framework/src/main/java/org/elasticsearch/test/ExternalNode.java

```diff
@@ -217,7 +217,7 @@ synchronized void stop() {
                 try {
                     process.waitFor();
                 } catch (InterruptedException e) {
-                    Thread.interrupted();
+                    Thread.currentThread().interrupt();
```

Comment:
Called in the `#stop()` method which can be interrupted when waiting for an external process to stop. I'd opt for rethrowing InterruptedException were we not implementing `Closeable` and delegating to `#stop()` in our implementation of `#close()`. Even then I'd just rethrow and restore the flag just in `#close()`. Client code does not need to be adapted at all as it is already aware of `InterruptedException` now.